sw/host/ot_certs/src/x509.rs

```diff
@@ -196,10 +196,36 @@ pub fn generate_certificate(tmpl: &template::Template) -> Result<Vec<u8>> {
     Ok(cert)
 }
 
-fn get_subject_alt_name(_x509: &X509) -> Result<IndexMap<AttributeType, Value<String>>> {
-    let empty: IndexMap<AttributeType, Value<String>> = Default::default();
+fn get_subject_alt_name(x509: &X509) -> Result<IndexMap<AttributeType, Value<String>>> {
+    let mut alt_names: IndexMap<AttributeType, Value<String>> = Default::default();
+    let Some(names) = x509.subject_alt_names() else {
+        return Ok(alt_names);
+    };
+    for general_name in names {
+        let type_ = unsafe { (*general_name.as_ptr()).type_ };
+        // We expect a directoryName.
+        ensure!(
+            type_ == openssl_sys::GEN_DIRNAME,
+            "only directory names are supported for subject alt names"
+        );
+        // SAFETY: OpenSSL's directoryName is represented by X509_NAME when the type
+        // is GEN_DIRNAME. It looks like this (only relevant parts shown):
+        // typedef struct GENERAL_NAME_st {
+        // #define GEN_DIRNAME     4
```

Comment:
how is `GEN_DIRNAM` definiton used in this example?

Reply:
I am not entirely sure I understand your question: I check that the type is `GEN_DIRNAME` at line 208?

Reply:
right, I just misread the comment.